src/tree.scss

```diff
@@ -1,183 +1,431 @@
+  white-space: normal;
+  overflow: visible;
+  text-overflow: initial;
+}
+
+@mixin tree-selected-state() {
+  background: var(--sapList_SelectionBackgroundColor);
+  border-bottom: $fd-tree-item-selected-border-bottom;
+}
+
+@mixin tree-focus() {
+  outline-style: dotted;
+  outline-width: var(--sapContent_FocusWidth);
+  outline-color: var(--sapContent_FocusColor);
+  outline-offset: -0.125rem;
+
+  @include fd-active() {
+    outline-color: var(--sapContent_ContrastFocusColor);
+  }
+}
+
+@mixin tree-states() {
+  cursor: pointer;
+
+  @include fd-hover() {
+    background: var(--sapList_Hover_Background);
+  }
+
+  @include fd-selected() {
+    @include tree-selected-state();
+
+    @include fd-hover() {
+      background: var(--sapList_Hover_SelectionBackground);
+    }
+  }
 
-  $fd-tree-header-color: #6a6d70 !default;
-  $fd-tree-header-background-color: #eeeeef !default;
-  $fd-tree-row-background-color--hover: #fafafa !default;
+  @include fd-active() {
+    background: var(--sapList_Active_Background);
 
-  $fd-tree-control-width: 18px !default;
-  $fd-tree-control-button-width-height: 34px !default;
-  $fd-tree-transition-params: 0.125s ease-in !default;
+    .#{$block}__expander,
+    .#{$block}__text,
+    .#{$block}__icon,
+    .has-navigation-indicator::after,
+    button::before {
+      color: var(--sapList_Active_TextColor);
+    }
+
+    @include fd-hover() {
+      background: var(--sapList_Active_Background);
+    }
+
+    @include fd-selected() {
+      background: var(--sapList_Active_Background);
+
+      @include fd-hover() {
+        background: var(--sapList_Active_Background);
+      }
+    }
 
+    @include fd-focus() {
+      outline-color: var(--sapContent_ContrastFocusColor);
+    }
+  }
+}
+
+// BLOCK
+.#{$block} {
   @include fd-reset();
-  @include fd-last-child();
 
-  position: relative;
   width: 100%;
-  max-width: 100%;
-  background-color: $fd-tree-background-color;
-  border-bottom-style: solid;
-  border-bottom-width: $fd-tree-border-width;
-  border-bottom-color: $fd-tree-border-color;
-  padding-left: 0;
-  padding-right: 0;
-  margin-bottom: 12px;
-  margin-left: 0;
-
-  &:last-child {
-    margin-bottom: 0;
-  }
-
-  &--header {
-    border-bottom: 0;
-    border-top-style: solid;
-    border-top-width: $fd-tree-border-width;
-    border-top-color: $fd-tree-border-color;
-    margin-bottom: 0;
-    background-color: $fd-tree-header-background-color;
-  }
-
-  &__group {
-    @include fd-reset();
+  list-style-type: none;
+
+  &[aria-hidden="true"] {
+    display: none;
+  }
 
-    transition: opacity $fd-animation-speed linear;
-    max-height: auto;
-    opacity: 1;
-    margin-bottom: 0;
-    margin-left: 0;
-    padding-left: 0;
-    padding-right: 0;
+  &.expanded-level-1 {
+    @include set-indentation(2, 1.5rem);
+  }
 
-    &.is-hidden {
-      display: none;
+  &.expanded-level-2 {
+    @include set-indentation(2, 1rem);
+    @include set-indentation(3, 2.5rem);
```

Comment:
![image](https://user-images.githubusercontent.com/26483208/89429760-3ed03c80-d73e-11ea-8025-e65c05478300.png)

Should Level 2 and level 3 without icons have that huge padding left?

Reply:
Not sure what you mean :) The padding that is between the expander and the icon is coming from the size of the icon container specified in list. Level 2 and Level 3 indentation is the same in all cases - with and without icons. When you collapse the level (for example collapse level 1) items content (including/excluding the icon) should be aligned.

Reply:
Sorry, I meant expander button :d. So there is additional `2.75rem` padding, even, when there is not expanding button on sides. I'm not sure if that's how it's supposed to be, docs don't show any image without expander buttons.

Reply:
![image](https://user-images.githubusercontent.com/26483208/89431534-5f00fb00-d740-11ea-81af-b469a94b01c9.png)
Ok, it looks stupid. Let's keep it as it is